src/array.js

```diff
@@ -46,8 +46,13 @@ inherits(ArraySchema, MixedSchema, {
     if (!this._typeCheck(value) || !this._subType) return value;
 
     let isChanged = false;
-    const castArray = value.map(v => {
-      const castElement = this._subType.cast(v, _opts);
+    const castArray = value.map((v, idx) => {
+      var path = makePath`${_opts.path}[${idx}]`;
+      var innerOptions = {
```

Comment:
can you move this inline below: `cast(v, { ..._opts, path: makePat`

Reply:
Thanks for quick check @jquense ! Made the change.